Forge.TreeWalker/src/ExpressionExecutor.cs

```diff
@@ -23,19 +25,36 @@ namespace Microsoft.Forge.TreeWalker
     public class ExpressionExecutor
     {
         /// <summary>
-        /// List of external type dependencies needed to compile expressions.
+        /// Code that Forge uses to prime the parentScript on initialization.
+        /// Evaluating some code moves roughly 600ms from the first expression to initializing the parentScript.
         /// </summary>
-        private List<Type> dependencies;
+        public static string ParentScriptCode = "(1+1).ToString()";
 
         /// <summary>
-        /// Script cache used to cache and re-use compiled Roslyn scripts.
+        /// The parentScriptTask kicks off initializing the Roslyn parentScript asynchronously, allowing ExpressionExecutor to construct very quickly.
+        /// Initializing the Roslyn parentScript takes about 2 seconds. This time is saved if the application takes time to initialize before the first WalkTree call.
         /// </summary>
-        private ConcurrentDictionary<string, Script<object>> scriptCache;
+        public Task parentScriptTask { get; private set; }
+
+        /// <summary>
+        /// The parentScript that, upon initialization, gets Created, RunAsync, and added to the ScriptCache.
+        /// All expressions that get Executed are continued from the parentScript, avoiding additional compiles.
+        /// 
+        /// Before improvement - each unique expression would cost 25MB, take 2 seconds on first execution, then 0ms on further executions.
+        /// After improvement - only the parentScript costs 25MB and takes 2 seconds. Each unique expression costs < 0.5MB, takes 15ms on first execution, then 0ms on further executions.
+        /// </summary>
+        private Script<object> parentScript;
+
+        /// <summary>
+        /// List of external type dependencies needed to compile expressions.
+        /// </summary>
+        private List<Type> dependencies;
 
         /// <summary>
-        /// Roslyn script options.
+        /// The ScriptCache is used to cache and re-use Roslyn scripts.
+        /// Scripts include the parentScript, as well as all unique expressions that get Executed.
         /// </summary>
```

Comment:
></summary> [](start = 12, length = 10)

consider to add your comment in the code review description:
ScriptCache now caches Scripts that come from the parentScript.ContinueWith. ScriptCache still caches one Script per unique Expression, but these no longer get compiled.

Reply:
Added more detail to ScriptCache description.

---
In reply to: [470870229](https://github.com/microsoft/Forge/pull/25#discussion_r470870229) [](ancestors = 470870229)